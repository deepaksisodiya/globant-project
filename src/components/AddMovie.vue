<template>
  <div>
    <form @submit="checkForm">
      <p v-if="errors.length">
        <b>Please correct the following error(s):</b>
        <ul>
          <li v-for="(error, index) in errors" :key="index">{{ error }}</li>
        </ul>
      </p>
      <p>
        <label for="movieName">Movie Name* </label>
        <input v-model="movieName" type="text" />
      </p>
      <p>
        <label for="genre">Genre*</label>
        <select v-model="genre">
          <option>Star Wars</option>
          <option>Vanilla Sky</option>
          <option>Atomic Blonde</option>
        </select>
      </p>
      <p>
        <label for="rating">Rating* </label>
        <input v-model="rating" type="number" min="0" max="5" />
      </p>
      <p>
        <label for="website">Website </label>
        <input v-model="website" type="text" />
      </p>
      <p>
        <input type="submit" value="Submit" />
      </p>
    </form>
  </div>
</template>

<script>
export default {
  name: "AddMovie",
  data() {
    return {
      errors: [],
      movieName: null,
      genre: null,
      rating: null,
      website: null
    };
  },
  props: {
    addNewMovie: {
      type: Function,
      required: true,
    }
  },
  methods: {
    checkForm: function(event) {
      event.preventDefault();
      this.errors = [];

      if (!this.movieName) {
        this.errors.push("Movie name is required");
      }

      if (!this.genre) {
        this.errors.push("genre is required");
      }

      if (!this.rating) {
        this.errors.push("rating is required");
      }

      if (!this.errors.length) {
        this.addMovieToList();
      }
    },

    addMovieToList() {
      const data = {
        movieName: this.movieName,
        genre: this.genre,
        rating: this.rating,
        website: this.website
      };
      this.addNewMovie(data);
      this.resetData();
    },

    resetData() {
      this.movieName = null;
      this.genre = null;
      this.rating = null;
      this.website = null;
    }
  }
};
</script>

<style>
input[type=text], select {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
}

input[type=submit] {
  width: 100%;
  background-color: #4CAF50;
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

input[type=submit]:hover {
  background-color: #45a049;
}

div {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
}
</style>
