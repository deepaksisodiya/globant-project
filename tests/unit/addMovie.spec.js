import { shallowMount } from "@vue/test-utils";
import AddMovie from "@/components/AddMovie.vue";

describe("AddMovie.vue", () => {
  it("should be an vue instace", () => {
    const addNewMovie = jest.fn();
    const wrapper = shallowMount(AddMovie, {
      propsData: { addNewMovie }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("should match snapshot", () => {
    const addNewMovie = jest.fn();
    const wrapper = shallowMount(AddMovie, {
      propsData: { addNewMovie }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should check initial state", () => {
    const addNewMovie = jest.fn();
    const wrapper = shallowMount(AddMovie, {
      propsData: { addNewMovie }
    });
    expect(wrapper.vm.movieName).toBe(null);
    expect(wrapper.vm.genre).toBe(null);
    expect(wrapper.vm.rating).toBe(null);
    expect(wrapper.vm.website).toBe(null);
  });

  it("should add movie to list", () => {
    const addNewMovie = jest.fn();
    const wrapper = shallowMount(AddMovie, {
      propsData: { addNewMovie }
    });
    expect(wrapper.vm.movieName).toBe(null);
    expect(wrapper.vm.genre).toBe(null);
    expect(wrapper.vm.rating).toBe(null);
    expect(wrapper.vm.website).toBe(null);
    wrapper.setData({
      movieName: "kaho na pyar he",
      genre: "romantic",
      rating: 5,
      website: "www.random.com"
    });
    expect(wrapper.vm.movieName).toBe("kaho na pyar he");
    wrapper.vm.addMovieToList();
    expect(addNewMovie).toHaveBeenCalled();
    expect(wrapper.vm.movieName).toBe(null);
  });

  it("should reset data", () => {
    const addNewMovie = jest.fn();
    const wrapper = shallowMount(AddMovie, {
      propsData: { addNewMovie }
    });
    wrapper.setData({
      movieName: "kaho na pyar he",
      genre: "romantic",
      rating: 5,
      website: "www.random.com"
    });
    wrapper.vm.resetData();
    expect(wrapper.vm.movieName).toBe(null);
    expect(wrapper.vm.genre).toBe(null);
    expect(wrapper.vm.rating).toBe(null);
    expect(wrapper.vm.website).toBe(null);
  });
});
