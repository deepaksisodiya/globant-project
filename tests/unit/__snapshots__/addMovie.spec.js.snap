// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie.vue should match snapshot 1`] = `
<div>
  <form>
    <!---->
     
    <p>
      <label
        for="movieName"
      >
        Movie Name* 
      </label>
       
      <input
        type="text"
      />
    </p>
     
    <p>
      <label
        for="genre"
      >
        Genre*
      </label>
       
      <select>
        <option>
          Action
        </option>
         
        <option>
          Adventure
        </option>
         
        <option>
          Comedy
        </option>
         
        <option>
          Crime
        </option>
         
        <option>
          Drama
        </option>
      </select>
    </p>
     
    <p>
      <label
        for="rating"
      >
        Rating* 
      </label>
       
      <input
        max="5"
        min="0"
        type="number"
      />
    </p>
     
    <p>
      <label
        for="website"
      >
        Website 
      </label>
       
      <input
        type="text"
      />
    </p>
     
    <p>
      <input
        type="submit"
        value="Submit"
      />
    </p>
  </form>
</div>
`;
